### PR TITLE
chore: checking codecov target

### PR DESCRIPTION
**Pull request by gsk967**
## Description

closes: #XXXX

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -44,7 +45,9 @@ ignore:
   - "*.md"
   - "*.rst"
   - "**/*.pb.go"
+  - "**/*.pb.gw.go"
   - "types/*.pb.go"
+  - "types/*.pb.gw.go"
```
Comment (toteki) [inline codecov.yml:50]: Are these patterns (types and **) both required?

Comment (gsk967) [reply codecov.yml]: No need, I will remove it

File: codecov.yml [hunk 2]
```diff
@@ -7,47 +7,42 @@ coverage:
     # Learn more at https://docs.codecov.io/docs/commit-status
     project:
       default:
-        threshold: 1% # allow this much decrease on project
+        target: auto
+        threshold: 0%
```
Comment (robert-zaremba) [inline codecov.yml:11]: why setting to 0% here?

File: util/store/store_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+// 	key := []byte("uint32")
+// 	val := sdk.AccAddress("1234")
+// 	// set uint32
+// 	err := SetAddress(store, key, val, "no error")
+// 	assert.NilError(t, err)
+
+// 	// get uint32
+// 	v := GetAddress(store, key, "no error")
+// 	assert.DeepEqual(t, v, val)
+// }
```
Comment (robert-zaremba) [inline util/store/store_test.go:76]: why commenting this out?

File: codecov.yml [hunk 4]
```diff
@@ -7,47 +7,42 @@ coverage:
...
       modules:
         target: 70%
-        flags:
-          - modules
+        paths:
+          - "x/"
+          - "!x/**/client/" # ignore client package
       client:
-        flags:
+        paths:
```
Comment (robert-zaremba) [inline codecov.yml:24]: let's add a target for the client as well

File: codecov.yml (hunk 2 above)
Comment (gsk967) [reply codecov.yml]: adding 1%

File: codecov.yml (hunk 4 above)
Comment (gsk967) [reply codecov.yml:25]: set the target to 70% for client

File: codecov.yml [hunk 5]
```diff
@@ -7,47 +7,43 @@ coverage:
     # Learn more at https://docs.codecov.io/docs/commit-status
     project:
       default:
-        threshold: 1% # allow this much decrease on project
+        target: auto
+        threshold: 1% # # allow this much decrease on project
```
Comment (robert-zaremba) [inline codecov.yml:11]:
```suggestion
        threshold: 35% # # allow this much decrease on project
```

File: x/oracle/keeper/historic_avg_test.go [hunk 6]
```diff
@@ -1,272 +1,267 @@
...
+// 	sumDec := sdk.MustNewDecFromStr(sum)
+// 	c, err := k.getCounter(denom, idx)
+// 	require.NoError(t, err)
+// 	tassert.Equal(t, sumDec, c.Sum, "sum idx=%d", idx)
+// 	tassert.Equal(t, num, c.Num, "num idx=%d", idx)
+// }
+
+// func newAvgCounter(sum, num uint32, start time.Time) types.AvgCounter {
+// 	return types.AvgCounter{Sum: sdk.NewDec(int64(sum)), Num: num, Start: start}
+// }
```
Comment (robert-zaremba) [inline x/oracle/keeper/historic_avg_test.go:267]: why this is commented out?

File: codecov.yml (hunk 5 above)
Comment (gsk967) [reply codecov.yml:11]: Why 35% 👀, i think this percentage is too much

File: x/oracle/keeper/historic_avg_test.go (hunk 6 above)
Comment (gsk967) [reply x/oracle/keeper/historic_avg_test.go]: I added previously to test the codecov, I will remove it

File: codecov.yml (hunk 5 above)
Comment (robert-zaremba) [reply codecov.yml:11]: Why it's too much?

**Review by robert-zaremba — APPROVED** [review]
pre-approving, let's increase the 1%

File: codecov.yml (hunk 5 above)
Comment (gsk967) [reply codecov.yml:11]: Okay, I am making 2%

File: codecov.yml [hunk 7]
```diff
@@ -7,47 +7,43 @@ coverage:
...
   - "*.rst"
   - "**/*.pb.go"
-  - "types/*.pb.go"
+  - "**/*.pb.gw.go"
   - "tests/*"
   - "tests/**/*"
   - "x/**/*.pb.go"
+  - "x/**/*.pb.gw.go"
+  - "x/**/types/*.pb.go"
+  - "x/**/types/*.pb.gw.go"
```
Comment (toteki) [inline codecov.yml:45]: Some lines are redundant:

To ignore all of a file type, no matter where:
`**.*/.pb.go`

To ignore a directory and all its contents:
`tests`

**Pull request merged**